### PR TITLE
Refactor/add more types 2

### PR DESCRIPTION
**Pull request by KristianKjerstad**
## What does this pull request change?
* define more types for SearhPage, EditSimaApplicaitonInput and some other small changes
* requires https://github.com/equinor/data-modelling-storage-service/pull/253

## Why is this pull request needed?
more error catching
## Issues related to this change

related to https://github.com/equinor/data-modelling-tool/issues/1359

### CODE REVIEW

File: web/packages/plugins/dmt-app/src/pages/SearchPage.tsx [hunk 1]
```diff
@@ -421,9 +450,11 @@ function SelectDataSource(props: {
       <MultiSelect
         label="Select data sources to search"
         initialSelectedItems={selectedDataSources}
-        handleSelectedItemsChange={(event: any) =>
-          setDataSources(event.selectedItems)
-        }
+        handleSelectedItemsChange={(event) => {
```
Comment (netr0m) [inline web/packages/plugins/dmt-app/src/pages/SearchPage.tsx:453]:
```suggestion
        handleSelectedItemsChange={(event) => {
```
Can/should we include a type here?

File: web/packages/plugins/sima-application-input/src/EditSimaApplicationInput.tsx [hunk 2]
```diff
@@ -60,22 +62,24 @@ const ReadOnlyTextField = (props: { text: string; label: string }) => {
   )
 }
 
-const STaskBlueprint = 'AnalysisPlatformDS/Blueprints/STask'
+const STaskBlueprint =
+  "Ansole.log('tabData', tabDataalysisPlatformDS/Blueprints/STask"
```
Comment (netr0m) [inline web/packages/plugins/sima-application-input/src/EditSimaApplicationInput.tsx:66]:
```suggestion
const STaskBlueprint = "AnalysisPlatformDS/Blueprints/STask"
```
fix typo / accidental paste

Comment (netr0m) [inline web/packages/plugins/sima-application-input/src/EditSimaApplicationInput.tsx:66]:
```suggestion
const STaskBlueprint: string = "AnalysisPlatformDS/Blueprints/STask"
```
fix typo, add type?

**Review by netr0m — CHANGES_REQUESTED** [review]
See the two inline comments; one for a typo in a variable's value and another for a potentially missing type. Otherwise it looks good!

File: web/packages/plugins/dmt-app/src/pages/SearchPage.tsx [hunk 3]
```diff
@@ -447,16 +447,17 @@ function SelectDataSource(props: {
         marginTop: '10px',
       }}
     >
-      <MultiSelect
-        label="Select data sources to search"
-        initialSelectedItems={selectedDataSources}
-        handleSelectedItemsChange={(event) => {
+      <Autocomplete
```
Comment (KristianKjerstad) [inline web/packages/plugins/dmt-app/src/pages/SearchPage.tsx:450]: MultiSelect from EDS is depreciated. Substituted with the Autocomplete component as recomended by EDS
https://eds-storybook-react.azurewebsites.net/?path=/docs/inputs-select-multiselect--default

**Review by netr0m — APPROVED** [review]

**Pull request merged**